src/formatDate.js

```diff
@@ -50,7 +50,27 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatedDate = date.split(fromFormat[3]);
+  const objestDate = {};
+  const result = [];
+
+  for (let i = 0; i < 3; i++) {
+    objestDate[fromFormat[i]] = formatedDate[i];
+  };
+
+  if (objestDate.YY >= 22) {
+    objestDate.YYYY = 19 + objestDate.YY;
+  } else if (objestDate.YY <= 22) {
+    objestDate.YYYY = 20 + objestDate.YY;
+  } else {
+    objestDate.YY = objestDate.YYYY.slice(2);
+  };
```

Comment:
За умовою завдання треба порівнювати з 30, а ти порівнюєш з 22, " When converting from YY to YYYY use 20YY if YY **< 30** and 19YY otherwise."
Але в цілому - гарний код)